### PR TITLE
COL-509 Verify asset previews available before exporting whiteboard

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-509

### CODE REVIEW

**Comment by paulkerschen** [comment]
Note that the code in the migration script substantially duplicates new code in the whiteboards API. It had to be picked out rather than called via API to avoid making the script dependent on a running  Express server.

Changes in the assets API are just to add in more explicit callback handling.

File: node_modules/col-whiteboards/lib/api.js [hunk 1]
```diff
@@ -1441,74 +1463,156 @@ var getWhiteboardAsPngFile = function(whiteboard, callback) {
...
+      }, 'Whiteboard is not exportable');
+      return callback({'code': 500, 'msg': 'The whiteboard cannot be exported: images have not finished processing'});
+    }
 
-  // Once the PNG has been generated (or the script fails), return to the caller
-  childProcess.on('close', function(code) {
-    if (code !== 0) {
+    // Generating a PNG version of a whiteboard is a very CPU intensive process that can hold up the
+    // Node.JS event loop. On top of that, fabric.js and its dependency node-canvas suffer from memory
+    // leaks when importing images onto a Cavas. To avoid running out of memory or blocking the event loop,
```
Comment (johncrossman) [inline node_modules/col-whiteboards/lib/api.js]: @paulkerschen perhaps we take this opportunity to fix `Cavas`?

**Comment by johncrossman** [comment]
Very nice, @paulkerschen  It seems that exporting a PNG is, in fact, rocket science.  One spelling fix then I'll merge.

**Comment by johncrossman** [comment]
🇪🇸  🍷

**Pull request merged**